src/TrackRowInfo.js

```diff
@@ -41,46 +43,107 @@ export default function TrackRowInfo(props) {
     } = props;
 
     // Dimensions
+    const defaultUnitWidth = 100;
+    const marginForMouseEvent = 20;
     const isLeft = rowInfoPosition === "left";
     const top = trackY;
-    const unitWidth = 100;
-    const width = unitWidth * rowInfoAttributes.length;
+    let unitWidths = [];    // Store default unit width for each vertical tracks
+    rowInfoAttributes.forEach((fieldInfo) => {
+        unitWidths.push({
+            field: fieldInfo.field,
+            type: fieldInfo.type,
+            width: defaultUnitWidth
+        });
+    });
+    const [widths, setWidths] = useState(unitWidths);
+    const totalWidth = d3.sum(widths.map(d => d.width));
     const height = trackHeight;
-    const left = isLeft ? trackX - width : trackX + trackWidth;
+    const left = isLeft ? trackX - totalWidth : trackX + trackWidth;
 
+    const divRef = useRef();
+    const resizerRef = useRef([...Array(rowInfoAttributes.length)].map(() => createRef()));
+    const [resizingIndex, setResizingIndex] = useState(-1);
+    
     // Determine position of each dimension.
-    let trackProps = [], xDomain = [], xRange = [];
+    let trackProps = [], currentLeft = 0;
     rowInfoAttributes.forEach((attribute, i) => {
         const fieldInfo = isLeft ? rowInfoAttributes[rowInfoAttributes.length - i - 1] : attribute;
-        let currentLeft = unitWidth * i;
+        const width = widths.find(d => d.field === fieldInfo.field && d.type === fieldInfo.type).width;
 
         trackProps.push({
-            left: currentLeft, top: 0, width: unitWidth, height: height,
+            left: currentLeft, top: 0, width, height,
             fieldInfo,
             isLeft
         });
-
-        // Domain and range for mouse event.
-        xDomain.push(currentLeft + unitWidth);
-        xRange.push(fieldInfo.field);
+        currentLeft += width;
     });
+    
```

Comment:
If you want to move it to another file that sounds good, but I am fine with it staying here for now since the width calculation is basically the only logic going on in the TrackRowInfo component at the moment.

Reply:
In that case, I will just leave the code as it is. Let's figure out in the future when we add more codes to TrackRowInfo.